### PR TITLE
update windows to ruby 2.7

### PR DESCRIPTION
**Pull request by pfrcks**

### CODE REVIEW

File: kubernetes/linux/setup.sh [hunk 1]
```diff
@@ -79,11 +79,3 @@ sudo apt-get remove ruby2.7-dev gcc make -y
 # Remove settings for cron.daily that conflict with the node's cron.daily. Since both are trying to rotate the same files
 # in /var/log at the same time, the rotation doesn't happen correctly and then the *.1 file is forever logged to.
 rm /etc/logrotate.d/alternatives /etc/logrotate.d/apt /etc/logrotate.d/azure-mdsd /etc/logrotate.d/rsyslog
-
-#Remove gemfile.lock for http_parser gem 0.6.0
-#see  - https://github.com/fluent/fluentd/issues/3374 https://github.com/tmm1/http_parser.rb/issues/70
-if [  -e "/var/lib/gems/2.6.0/gems/http_parser.rb-0.6.0/Gemfile.lock" ]; then
```
Comment (ganga1980) [inline kubernetes/linux/setup.sh:85]: is this fixed in 2.7 ?

Comment (ganga1980) [inline kubernetes/linux/setup.sh:85]: is this fixed in 2.7 ?

File: kubernetes/windows/Dockerfile [hunk 2]
```diff
@@ -9,8 +9,8 @@ ARG IMAGE_TAG=win-ciprod03172022
 # Docker creates a layer for every RUN-Statement
 RUN powershell -Command "Set-ExecutionPolicy Bypass -Scope Process -Force; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))"
 # Fluentd depends on cool.io whose fat gem is only available for Ruby < 2.5, so need to specify --platform ruby when install Ruby > 2.5 and install msys2 to get dev tools
-RUN choco install -y ruby --version 2.6.5.1 --params "'/InstallDir:C:\ruby26'" \
-&& choco install -y msys2 --version 20211130.0.0 --params "'/NoPath /NoUpdate /InstallDir:C:\ruby26\msys64'" \
+RUN choco install -y ruby --version 2.7.5.1 --params "'/InstallDir:C:\ruby27'" \
+&& choco install -y msys2 --version 20211130.0.0 --params "'/NoPath /NoUpdate /InstallDir:C:\ruby27\msys64'" \
```
Comment (ganga1980) [inline kubernetes/windows/Dockerfile:13]: is this validated with Ruby2.7 for windows?

Comment (ganga1980) [inline kubernetes/windows/Dockerfile:13]: is this validated with Ruby2.7 for windows?

File: kubernetes/linux/setup.sh (hunk 1 above)
Comment (pfrcks) [reply kubernetes/linux/setup.sh:85]: yes

Comment (pfrcks) [reply kubernetes/linux/setup.sh:85]: yes

File: kubernetes/windows/Dockerfile (hunk 2 above)
Comment (pfrcks) [reply kubernetes/windows/Dockerfile:13]: yes

Comment (pfrcks) [reply kubernetes/windows/Dockerfile:13]: yes

**Review by ganga1980 — APPROVED** [review]

**Pull request merged**